### PR TITLE
Mve2 pong api angle behaviour

### PR DESCRIPTION
**Pull request by opaleboreale**

### CODE REVIEW

File: back/transcendence_django/back_game/game_entities/ball.py [hunk 1]
```diff
@@ -48,13 +48,26 @@ def update_collision(self, paddle):
    def update_position(self, new_position):
       for paddle in self.paddles:
          if self.is_paddle_collision(self.position, paddle):
+            # side = self.get_collision_side(new_position, paddle)
```
Comment (romann-broque) [inline back/transcendence_django/back_game/game_entities/ball.py:51]: remove comments

File: back/transcendence_django/back_game/game_entities/ball.py [hunk 2]
```diff
@@ -48,13 +48,26 @@ def update_collision(self, paddle):
...
             self.__push_ball(paddle)
             new_position = self.position
+            collision_point = self.get_collision_point(paddle)
+            self.speed = paddle.calc_speed_after_collision(collision_point)
+            logger.info(f"New speed is: ({self.speed.x}, {self.speed.y})")
+            self.position = new_position
             break
             # return # Exit the function after handling the collision
 
       # If no collision with paddles, proceed with the normal update
```
Comment (romann-broque) [inline back/transcendence_django/back_game/game_entities/ball.py:63]: remove comments

File: back/transcendence_django/back_game/game_entities/ball.py [hunk 3]
```diff
@@ -48,13 +48,26 @@ def update_collision(self, paddle):
...
       self.__update_wall_collision(new_position)
+      # self.set_position(new_position)
```
Comment (romann-broque) [inline back/transcendence_django/back_game/game_entities/ball.py:65]: also there

**Review by romann-broque — CHANGES_REQUESTED** [review]

File: back/transcendence_django/back_game/game_entities/ball.py (hunk 1 above)
Comment (ethanolmethanol) [reply back/transcendence_django/back_game/game_entities/ball.py:51]: True that !

File: back/transcendence_django/back_game/game_entities/ball.py (hunk 2 above)
Comment (ethanolmethanol) [reply back/transcendence_django/back_game/game_entities/ball.py:63]: What he said !

File: back/transcendence_django/back_game/game_entities/ball.py (hunk 3 above)
Comment (ethanolmethanol) [reply back/transcendence_django/back_game/game_entities/ball.py:65]: Indeed !

File: back/transcendence_django/back_game/game_entities/paddle.py [hunk 4]
```diff
@@ -79,3 +89,17 @@ def update(self, config):
    def move(self, direction):
       self.rate = min(max(self.rate + self.speed * direction, 0), 1)
       self.__update_position()
+
+   def calc_speed_after_collision(self, collision_point):
```
[the remark this answers is not in the document]
Comment (ethanolmethanol) [reply back/transcendence_django/back_game/game_entities/paddle.py:93]: Get that bread !

File: back/transcendence_django/back_game/game_entities/paddle.py [hunk 5]
```diff
@@ -79,3 +89,17 @@ def update(self, config):
...
+      speed_component = self.calculate_speed_direction(collision_point)
+      u_speed = speed_component.unit_vector()
+      return Vector(INITIAL_BALL_SPEED_COEFF * u_speed.x, INITIAL_BALL_SPEED_COEFF * u_speed.y)
+
+   def calculate_speed_direction(self, collision_point):
```
[the remark this answers is not in the document]
Comment (ethanolmethanol) [reply back/transcendence_django/back_game/game_entities/paddle.py:98]: Get it !

File: back/transcendence_django/back_game/game_entities/paddle.py [hunk 6]
```diff
@@ -61,6 +63,14 @@ def __convert_rate_to_position(self, rate):
          self.axis['start'].y + (self.axis['end'].y - self.axis['start'].y) * rate
       ).round()
 
+   def __calculate_convexity_center(self):
```
[the remark this answers is not in the document]
Comment (ethanolmethanol) [reply back/transcendence_django/back_game/game_entities/paddle.py:66]: GETTT !

File: back/transcendence_django/back_game/game_settings/game_constants.py [hunk 7]
```diff
@@ -1,22 +1,30 @@
...
+GAME_REFRESH_RATE = 0.005
 
 # Game arena
 MIN_PLAYER = 2
 MAX_PLAYER = 10
 
 # # Ball parameters
-INITIAL_SPEEDX = 5
-INITIAL_SPEEDY = 5
+# BALL_INITIAL_POSITION = {'x': 50, 'y': 50}
```
[the remark this answers is not in the document]
Comment (ethanolmethanol) [reply back/transcendence_django/back_game/game_settings/game_constants.py:13]: BEGONE !

File: docker-compose.yml [hunk 8]
```diff
@@ -229,12 +227,12 @@ services:
 volumes:
     prometheus-data:
     grafana-data:
-    front-data:
-        driver: local
-        driver_opts:
-            type: 'none'
-            o: 'bind'
-            device: './front/dist/transcendence/browser'
+    # front-data:
```
[the remark this answers is not in the document]
Comment (ethanolmethanol) [reply docker-compose.yml:230]: Come back !

**Review by romann-broque — APPROVED** [review]

**Pull request merged**